### ISSUE TITLE
[CI] HttpClientTests#testHttpsWithTLSv1GeneratesWarnings failure on 6.8

### ISSUE THREAD

**Issue by cbuescher** [elastic/elasticsearch#72396]
Maybe related to use of zulu11.

**Build scan**:
https://gradle-enterprise.elastic.co/s/xo5d4267i2sx2

**Repro line**:

```
./gradlew ':x-pack:plugin:watcher:unitTest' \
  -Dtests.seed=948261F09DCAF954 \
  -Dtests.class=org.elasticsearch.xpack.watcher.common.http.HttpClientTests \
  -Dtests.method="testHttpsWithTLSv1GeneratesWarnings" \
  -Dtests.security.manager=true \
  -Dtests.locale=hi-IN \
  -Dtests.timezone=US/Aleutian \
  -Dcompiler.java=11 \
  -Druntime.java=11
```

**Reproduces locally?**:

No, but I didn't try zulu11 which is used in this matrix run yet.

**Applicable branches**:

6.8

**Failure history**:

Twice today, this is the second build scan: https://gradle-enterprise.elastic.co/s/cgopoijyhhhgg

**Failure excerpt**:

```
> Throwable #1: javax.net.ssl.SSLHandshakeException: Remote host terminated the handshake
   > 	at __randomizedtesting.SeedInfo.seed([F7C2BC6FDF2E1368:BF1A330EE5B19C98]:0)
   > 	at java.base/sun.security.ssl.SSLSocketImpl.handleEOF(SSLSocketImpl.java:1598)
   > 	at java.base/sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1426)
   > 	at java.base/sun.security.ssl.SSLSocketImpl.readHandshakeRecord(SSLSocketImpl.java:1324)
   > 	at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:440)
   > 	at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:411)
   > 	at org.apache.http.conn.ssl.SSLConnectionSocketFactory.createLayeredSocket(SSLConnectionSocketFactory.java:394)
   > 	at org.apache.http.conn.ssl.SSLConnectionSocketFactory.connectSocket(SSLConnectionSocketFactory.java:353)
   > 	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:141)
   > 	at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:353)
   > 	at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:380)
   > 	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:236)
   > 	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:184)
   > 	at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:88)
   > 	at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110)
   > 	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:184)
   > 	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:82)
   > 	at org.elasticsearch.xpack.watcher.common.http.HttpClient.lambda$execute$0(HttpClient.java:175)
   > 	at java.base/java.security.AccessController.doPrivileged(Native Method)
   > 	at org.elasticsearch.xpack.core.common.socket.SocketAccess.doPrivileged(SocketAccess.java:32)
   > 	at org.elasticsearch.xpack.watcher.common.http.HttpClient.execute(HttpClient.java:175)
   > 	at org.elasticsearch.xpack.watcher.common.http.HttpClientTests.testSslMockWebserver(HttpClientTests.java:316)
   > 	at org.elasticsearch.xpack.watcher.common.http.HttpClientTests.testHttpsWithTLSv1GeneratesWarnings(HttpClientTests.java:228)
   > 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
   > 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
   > 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
   > 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
   > 	at java.base/java.lang.Thread.run(Thread.java:829)
   > Caused by: java.io.EOFException: SSL peer shut down incorrectly
   > 	at java.base/sun.security.ssl.SSLSocketInputRecord.read(SSLSocketInputRecord.java:483)
   > 	at java.base/sun.security.ssl.SSLSocketInputRecord.readHeader(SSLSocketInputRecord.java:472)
   > 	at java.base/sun.security.ssl.SSLSocketInputRecord.decode(SSLSocketInputRecord.java:160)
   > 	at java.base/sun.security.ssl.SSLTransport.decode(SSLTransport.java:110)
   > 	at java.base/sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1418)
   > 	... 56 more

```

**Comment by iverase** [comment]
Another failure today: https://gradle-enterprise.elastic.co/s/fpjub6b6sd23q

**Comment by ywangd** [comment]
This failed a couple more times today with zulu8, zulu11 and adoptopenjdk8 adoptopenjdk11. It is likely some changes to TLS get backported and breaks certain assumption that we have.
* https://gradle-enterprise.elastic.co/s/72p6kdczfeqbi
* https://gradle-enterprise.elastic.co/s/uri5mphsdq5vk
* https://gradle-enterprise.elastic.co/s/ek2tocq5vd3is
* https://gradle-enterprise.elastic.co/s/jxptefcxqrzmy

**Comment by ywangd** [comment]
Indeed, TLSv1 is disabled in the latest releases of above JDKs. The error message is:
> javax.net.ssl.SSLHandshakeException: No appropriate protocol (protocol is disabled or cipher suites are inappropriate)

I will mute this test for now. It can probably be removed later on after discussion.

**Comment by ywangd** [comment]
Muted with 4d761633d552945c4cbde6e4cd510be3068007b8

**Comment by jkakavas** [comment]
This is specific only to 6.8 and already muted. I will close this issue as won't fix since 6.8 is EOL
